aws-s3-transfer-manager/tests/download_test.rs

```diff
@@ -420,3 +421,30 @@ async fn test_retry_max_attempts() {
     let requests = http_client.actual_requests().collect::<Vec<_>>();
     assert_eq!(4, requests.len());
 }
+
+/// Test the if_match header was added correctly based on the response from server.
+#[tokio::test]
+async fn test_download_if_match() {
```

Comment:
The tag will not change unless the object changed. 
I added a test for mock the object change, but the error will just be the service error from S3 that complains about if-match precondition didn't work.

In aws-c-s3, we override this error to our customized error code to inform it's because the object changed during download. But, in here, we are not overriding any service error, right?

Reply:
We haven't yet I think this is fine for now though unless you see an ergonomic reason to change it for users.

Reply:
The reason we override it for aws-c-s3 mostly because the if-match header was added by the client under the hood, so, it may be confusing to user for the if-match precondition failure. 
But, not a big deal for now.